src/cast_core/src/ExternalScriptRunner.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/ExternalScriptRunner.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/MongoException.hpp>
+#include <gennylib/context.hpp>
+
+#include <value_generators/DocumentGenerator.hpp>
+
+#include <iostream>
+#include <string>
+#include <cstdlib>
+#include <boost/filesystem.hpp>
+#include <cstdio>
+#include <stdexcept>
+#include <array>
+#include <stdio.h>
+#include <unistd.h>
+
+namespace genny::actor {
+
+struct ExternalScriptRunner::PhaseConfig {
+    std::string script;
+
+    std::string mongoServerURI;
+
+    // Record data retruned by the script
+    metrics::Operation operation;
+
+    // Record data of the script operation itself
+    metrics::Operation scriptOperation;
+
+    PhaseConfig(PhaseContext& phaseContext, ActorId id)
+        : script{phaseContext["Script"].to<std::string>()},
+          mongoServerURI{phaseContext["MongoServerURI"].maybe<std::string>().value_or("--nodb")},
+          operation{phaseContext.operation("DefaultMetricsName", id)},
+          scriptOperation{phaseContext.namedOperation("ExternalScript", id)} {}
```

Comment:
Let's just document what these metrics objects represent - here and/or in the docs yaml.